libs/application/ui-shell/src/utils.ts

```diff
@@ -69,3 +69,23 @@ export function extractAnswersToSubmitFromScreen(
       return pick(data, [screenId])
   }
 }
+
+export function isJSONObject(message?: string) {
+  if (!message) {
+    return undefined
+  }
+
+  return Boolean(message.match(/^\{.*\}/g))
+}
+
+export function parseMessage(message?: string) {
+  if (!message) {
+    return undefined
+  }
+
+  if (isJSONObject(message)) {
+    return JSON.parse(message)
```

Comment:
Wondering about this, from your test case above:

```javascript
it('return true if the message is an object', () => {
  expect(isJSONObject('{field:true}')).toBeTruthy()
})
```

If you were to run `JSON.parse('{field:true}')` you would get an error since it is not valid JSON.

Reply:
Good catch! will update that

Reply:
I don't know why I was doing this by myself when I can just use a try/catch around JSON.parse 🤦